hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -152,11 +152,12 @@ public void run() {
               addBatchToShippingQueue(batch);
             }
           }
-        } catch (IOException e) { // stream related
-          if (handleEofException(e, batch)) {
+        } catch (WALEntryFilterRetryableException | IOException e) { // stream related
+          if (e instanceof IOException && handleEofException((IOException) e, batch)) {
```

Comment:
Instead of adding checks for `e instanceof IOException &&`, we could change the argument for `handleEofException`  to accept Exception instead of IOException. The very first thing  `handleEofException` does is to check whether `e instance of EOFException or not`. If e is `WALEntryFilterRetryableException` then it will return false immediately.